src/main/kotlin/io/github/dockyardmc/Main.kt

```diff
@@ -83,23 +86,46 @@ fun main(args: Array<String>) {
             }
         }
     }
-    Commands.add("/tp") {
-        addArgument("world", StringArgument()) { listOf("main", "alt") }
-        execute {
-            val world = getArgument<String>("world")
-            if (world == "main") {
-                it.player?.teleport(WorldManager.mainWorld.defaultSpawnLocation)
-            } else if (world == "alt") {
-                it.player?.teleport(altWorld.defaultSpawnLocation)
-            }
-        }
-    }
 
-    val poolMain = EventPool.withFilter { it.context.contains(WorldManager.mainWorld) || it.context.isGlobalEvent }
-    val poolAlt = EventPool.withFilter { it.context.contains(altWorld) || it.context.isGlobalEvent }
+    val poolMain = EventPool.withFilter("mainworldpool") { it.context.contains(WorldManager.mainWorld) || it.context.isGlobalEvent }
+    val poolAlt = EventPool.withFilter("altworldpool") { it.context.contains(altWorld) || it.context.isGlobalEvent }
 
     poolMain.on<PlayerBlockBreakEvent> { it.cancelled = true }
     poolAlt.on<PlayerBlockBreakEvent> { DockyardServer.broadcastMessage("Player blocked in alt world.") }
 
+
+    val customPool = EventPool(parent = null, name = "custompool")
+    customPool.on<CustomEvent> { DockyardServer.broadcastMessage("[${it.value}] pool -> custom event! ${it.int++}") }
+
+    val subPool = EventPool(parent = customPool, name = "subpool-1")
```

Comment:
Probably want some better syntax for this, something like

`customPool.subPool("subpool-1")` instead of passing the parent into new instance parameter